tests/func-tests/node_placement_test.go

```diff
@@ -17,39 +22,146 @@ import (
 	kvtutil "kubevirt.io/kubevirt/tests/util"
 
 	networkaddonsv1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1"
-	"kubevirt.io/client-go/kubecli"
-	"kubevirt.io/kubevirt/tests/flags"
-
 	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+	"kubevirt.io/client-go/kubecli"
 )
 
-const nodePlacementSamplingSize = 20
-
-var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", func() {
+const (
+	nodePlacementSamplingSize = 20
+	hcoLabel                  = "node.kubernetes.io/hco-test-node-type"
+	infra                     = "infra"
+	workloads                 = "workloads"
+	group                     = "hco.kubevirt.io"
+	version                   = "v1beta1"
+	kind                      = "HyperConverged"
+	hcoResource               = "hyperconvergeds"
+	name                      = "kubevirt-hyperconverged"
+)
 
+var _ = Describe("[rfe_id:4356][crit:medium][vendor:cnv-qe@redhat.com][level:system]Node Placement", Ordered, func() {
 	var workloadsNode *v1.Node
 
 	tests.FlagParse()
 	client, err := kubecli.GetKubevirtClient()
 	kvtutil.PanicOnError(err)
 
-	workloadsNodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{
-		LabelSelector: "node.kubernetes.io/hco-test-node-type==workloads",
-	})
-	kvtutil.PanicOnError(err)
-
-	if workloadsNodes != nil && len(workloadsNodes.Items) == 1 {
-		workloadsNode = &workloadsNodes.Items[0]
+	// store the existing HCO CR obtained in BeforeALl and revert in AfterAll stage
+	originalHco := &hcov1beta1.HyperConverged{}
+
+	BeforeAll(func() {
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), k8smetav1.ListOptions{LabelSelector: "node-role.kubernetes.io/worker"})
+		kvtutil.PanicOnError(err)
+		totalNodes := len(nodes.Items)
+
+		// Label all but first node with "node.kubernetes.io/hco-test-node-type=infra"
+		// We are doing this to remove dependency of this Describe block on a shell script that
+		// labels the nodes this way
+		for i := 0; i < totalNodes; i++ {
+			err = setHcoNodeTypeLabel(client, &nodes.Items[i], infra)
+			kvtutil.PanicOnError(err)
+			if i == totalNodes-1 {
+				break
+			}
+		}
```

Comment:
I think this will be simpler:
```suggestion
		for _, node := range nodes.Items[:len(nodes.Items) - 1] {
			err = setHcoNodeTypeLabel(client, &node, infra)
			kvtutil.PanicOnError(err)
		}
```

WDYT?